sdk/storage/azfile/fileerror/error_codes.go

```diff
@@ -0,0 +1,103 @@
+//go:build go1.18
+// +build go1.18
+
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+package fileerror
+
+import (
+	"errors"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/Azure/azure-sdk-for-go/sdk/storage/azfile/internal/generated"
+)
+
+// HasCode returns true if the provided error is an *azcore.ResponseError
+// with its ErrorCode field equal to one of the specified Codes.
+func HasCode(err error, codes ...Code) bool {
+	var respErr *azcore.ResponseError
+	if !errors.As(err, &respErr) {
+		return false
+	}
+
+	for _, code := range codes {
+		if respErr.ErrorCode == string(code) {
+			return true
+		}
+	}
+
+	return false
+}
+
+// Code - Error codes returned by the service
+type Code = generated.StorageErrorCode
+
+const (
+	AccountAlreadyExists                 Code = "AccountAlreadyExists"
+	AccountBeingCreated                  Code = "AccountBeingCreated"
+	AccountIsDisabled                    Code = "AccountIsDisabled"
+	AuthenticationFailed                 Code = "AuthenticationFailed"
+	AuthorizationFailure                 Code = "AuthorizationFailure"
+	AuthorizationPermissionMismatch      Code = "AuthorizationPermissionMismatch"
+	AuthorizationProtocolMismatch        Code = "AuthorizationProtocolMismatch"
+	AuthorizationResourceTypeMismatch    Code = "AuthorizationResourceTypeMismatch"
+	AuthorizationServiceMismatch         Code = "AuthorizationServiceMismatch"
+	AuthorizationSourceIPMismatch        Code = "AuthorizationSourceIPMismatch"
+	CannotDeleteFileOrDirectory          Code = "CannotDeleteFileOrDirectory"
+	ClientCacheFlushDelay                Code = "ClientCacheFlushDelay"
+	ConditionHeadersNotSupported         Code = "ConditionHeadersNotSupported"
+	ConditionNotMet                      Code = "ConditionNotMet"
+	ContainerQuotaDowngradeNotAllowed    Code = "ContainerQuotaDowngradeNotAllowed"
```

Comment:
why is there a container error code here?

Reply:
This is coming from the generated code. Removed it.